File: JAVA/Swap two nibbles in a byte.java
class Solution {
    static int swapNibbles(int n) {
        String s=Integer.toBinaryString(n);
        String curr="",ans="";
        for(int i=s.length()-1;i>=0;i--){
            curr=s.charAt(i)+curr;
            if(curr.length()==4){
                ans+=curr;
                curr="";
            }
        }
        if(curr.length()!=0){
            while(curr.length()!=4) curr='0'+curr;
            ans+=curr;
        }
        while(ans.length()<8) ans+='0';
        return Integer.parseInt(ans,2);
    }
}

File: C++/Swap two nibbles in a byte.cpp
class Solution {
  public:
    int swapNibbles(int n) {
        int ans = 0;
        for(int i = 0; i < 8; i++) {
            if((n&1) == 1) {
                if(i < 4)
                    ans = ans + (1 << (4 + i));
                else 
                    ans = ans + (1 << (i - 4));
            }
            n = n >> 1;
        }
        return ans;
    }
};
